src/mlpack/core/data/scaler_methods/maxabsscaler.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file maxabsscaler.hpp
+ * @author Jeffin Sam
+ *
+ * MaxAbsScaler class to scale features.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_MAX_ABS_SCALE_HPP
+#define MLPACK_CORE_DATA_MAX_ABS_SCALE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+/**
+ * A simple MaxAbs Scaler class.
+ *
+ * Given an input dataset this class helps you to scale each
+ * feature by its maximum absolute value.
+ *
+ * @code
+ * arma::Mat<double> input = loadData();
+ * arma::Mat<double> output;
+ *
+ * // Scale the features.
+ * MaxAbsScaler<double> scale;
+ * scale.Tranform(input, output);
+ *
+ * // Retransform the input.
+ * scale.InverseTransform(output, input);
+ * @endcode
+ */
+class MaxAbsScaler
+{
+ public:
+  /**
+  * Function to scale Features.
+  *
+  * @param input Dataset to scale features.
+  * @param output Output matrix with scaled features.
+  */
+  template<typename MatType>
+  void Transform(const MatType& input, MatType& output)
+  {
+    output.copy_size(input);
+    itemMin = arma::min(input, 1);
+    itemMax = arma::max(input, 1);
+    scale = arma::max(arma::abs(itemMin), arma::abs(itemMax));
+    // Handline Zeroes in scale vector.
+    for (size_t i = 0; i < scale.n_elem; i++)
+    {
+      if (scale(i) == 0)
+      {
+        scale(i) = 1;
+      }
+    }
+    for (size_t i = 0; i < input.n_rows; i++)
+    {
+      for (size_t j = 0; j < input.n_cols; j++)
+      {
+        output(i, j) = input(i, j) / scale(i);
```

Comment:
We could simplify this by using `.each_col()`:

```cpp
output = input.each_col() / scale.
```

maybe we have to use `% 1.0 / scale` here.

A similar approach should work for the other classes as well. I can comment on the other classes as well, if you like.

Reply:
@zoq , That was a really beautiful way of reducing lines. Thanks for pointing it out. I have changed it for other class as well.

Although I didn't undestand the difference between `output = input.each_col() / scale`  and `output = input.each_col() % 1.0 / scale`, but i have used the latter one everywhere.

Reply:
Should be the same.

Reply:
Ok : ) , Made changes everywhere.